contracts/escrow/dapps/PoolTogetherFacet.sol

```diff
@@ -68,18 +66,17 @@ contract PoolTogetherFacet is PausableMods, DappMods {
 
         address ticketAddress = PoolTogetherLib.getTicketAddress(tokenAddress);
         uint256 balanceBefore = LibEscrow.balanceOf(loanID, ticketAddress);
-        IERC20(tokenAddress).safeApprove(address(prizePool), amount);
-
-        bytes memory callData =
-            abi.encode(
-                PrizePoolInterface.depositTo.selector,
-                address(this),
-                amount,
-                ticketAddress,
-                address(this)
-            );
-        LibDapps.s().loanEscrows[loanID].callDapp(address(prizePool), callData);
 
+        // Set token allowance to Prize Pool
+        LibEscrow.e(loanID).setTokenAllowance(tokenAddress, address(prizePool));
+        bytes memory callData = abi.encodeWithSelector(
+            PrizePoolInterface.depositTo.selector,
+            address(LibEscrow.e(loanID)), // Tickets sent to Escrow
+            amount,
+            ticketAddress,
+            address(this) // Referrer is the Teller Diamond
+        );
+        LibDapps.s().loanEscrows[loanID].callDapp(address(prizePool), callData);
```

Comment:
should use `LibEscrow.e(loanID)` just to stay consistent

Reply:
I like `escrow` more ;)